presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -47,15 +47,19 @@ public TimestampDecoder(ConnectorSession session, String path)
     public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
     {
         DocumentField documentField = hit.getFields().get(path);
-        if (documentField == null) {
-            output.appendNull();
+        Object value = getter.get();
```

Comment:
We should call `getter.get()` only if `documentField` is null. It may have non-trivial performance impact.

Reply:
I'll update tonight and push again!